src/models/syncRule.ts

```diff
@@ -133,11 +153,146 @@ export default class SyncRule {
+                },
+            }),
+            {}
+        );
+
+        return SyncRule.build({
+            ...this.syncRule,
+            builder: {
+                ...this.syncRule.builder,
+                useDefaultIncludeExclude: false,
+                metadataIncludeExcludeRules,
+            },
+        });
+    }
+
+    public moveRuleFromExcludeToInclude(type: string, ruleIndexes: number[]): SyncRule {
+        return this.moveIncludeExcludeRules(type, ruleIndexes, true);
+    }
+
+    public moveRuleFromIncludeToExclude(type: string, ruleIndexes: number[]): SyncRule {
+        return this.moveIncludeExcludeRules(type, ruleIndexes, false);
+    }
+
+    private moveIncludeExcludeRules(
+        type: string,
+        ruleIndexes: number[],
+        include: boolean
+    ): SyncRule {
+        if (!this.metadataExcludeIncludeRules) {
+            throw Error("metadataExcludeIncludeRules is not defined");
+        }
+
+        const oldIncludeRules = this.metadataExcludeIncludeRules[type].includeRules;
+        const oldExcludeRules = this.metadataExcludeIncludeRules[type].excludeRules;
+
+        if (include) {
+            const rulesToInclude = oldExcludeRules.filter((_, index) =>
+                ruleIndexes.includes(index)
+            );
+
+            const rulesToIncludeWithParents = _.uniq(
+                rulesToInclude.reduce(
+                    (array: string[], rule: string) => [
+                        ...array,
+                        rule,
+                        ...extractParentsFromRule(rule),
+                    ],
+                    []
+                )
+            );
+
+            const excludeIncludeRules = {
+                includeRules: _.uniq([...oldIncludeRules, ...rulesToIncludeWithParents]),
+                excludeRules: oldExcludeRules.filter(
+                    rule => !rulesToIncludeWithParents.includes(rule)
+                ),
+            };
+
+            return this.updateIncludeExcludeRules(type, excludeIncludeRules);
+        } else {
+            const rulesToExclude = oldIncludeRules.filter((_, index) =>
+                ruleIndexes.includes(index)
+            );
+
+            const childrenFromRule = (rule: string) =>
+                oldIncludeRules.filter((children: string) => children.includes(rule + "."));
+
+            const rulesToExcludeWithChildren = _.uniq(
+                rulesToExclude.reduce(
+                    (array: string[], rule: string) => [...array, rule, ...childrenFromRule(rule)],
+                    []
+                )
+            );
+
+            const excludeIncludeRules = {
+                includeRules: oldIncludeRules.filter(
+                    rule => !rulesToExcludeWithChildren.includes(rule)
+                ),
+                excludeRules: [...oldExcludeRules, ...rulesToExcludeWithChildren],
+            };
+
+            return this.updateIncludeExcludeRules(type, excludeIncludeRules);
+        }
+    }
```

Comment:
Following https://github.com/EyeSeeTea/metadata-synchronization/pull/323/files#r371679285

If we took the rules instead of its indexes we could remove the ``includes(index)``. Also I believe that we do not even need the ``rulesToInclude`` and ``rulesToExclude`` filter as these should arrive as parameters now.

Helper method ``childrenFromRule`` could be moved to utils same as ``extractParentsFromRule``.

Sintax improvement:
```
const { includeRules: oldIncludeRules, excludeRules: oldExcludeRules } = this.metadataExcludeIncludeRules[type];
```

Reply:
So, do I make this change now or when we move this feature to development-malaria?

Reply:
We can wait, no worries.